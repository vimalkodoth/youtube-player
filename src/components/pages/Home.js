import React from "react";
import { css } from "@emotion/core";
import Form from '../Form';

function Home() {
    return (
        <div css={Wrapper}>
            <Form to={
                (url) => {
                    return {
                        pathname: `/player/${url}`,
                        state: {}
                    }
                }
            } />
        </div>
    );
}

export default Home;

const Wrapper = css`
    display: flex;
    height: 100vh;
    justify-content: center;
    align-items: center;
    border: 1px solid #fff;
`;
