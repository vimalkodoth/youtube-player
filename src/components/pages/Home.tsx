import React from "react";
import { css } from "@emotion/core";
import Form from '../Form';

const Home: React.FC = () => {
    return (
        <div css={Wrapper}>
            <Form to={
                (url: any) => {
                    return {
                        pathname: `/player/${url}`,
                        state: {
                            url
                        }
                    }
                }
            } />
        </div>
    );
}

export default Home;

const Wrapper = css`
    display: flex;
    height: 100vh;
    justify-content: center;
    align-items: center;
    border: 1px solid #fff;
`;
