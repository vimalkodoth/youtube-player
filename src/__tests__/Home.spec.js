import React from "react";
import { render } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import Home from "./../components/pages/Home";

let store;
const mockStore = configureStore({});
const originalConsoleError = console.error;

describe("Home", () => {
    beforeEach(() => {
        console.error = jest.fn();
        store = mockStore({});
    });

    afterEach(() => {
        console.error = originalConsoleError;
        store = mockStore({});
    });

    it("renders correctly", () => {
        const component = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Home />
                </MemoryRouter>
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
});
