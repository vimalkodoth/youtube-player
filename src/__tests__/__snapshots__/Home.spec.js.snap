// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  class="css-1kjoe0h-Wrapper"
>
  <div
    class="css-eyw9vr-ListStyle"
  >
    <form>
      <label
        for="url"
      >
        Enter URL:
      </label>
      <br />
      <input
        class="css-2q23bc-Input"
        name="url"
        type="text"
      />
      <a
        href="/player/undefined"
      >
        <input
          class="css-2q23bc-Input"
          type="submit"
          value="Play"
        />
      </a>
    </form>
  </div>
</div>
`;
